darwin/importer/importer.py

```diff
@@ -1012,30 +1051,68 @@ def process_local_file(local_path):
                 for file in tqdm(
                     files_to_track, desc="Importing annotations from local file"
                 ):
-                    import_annotation(file)
+                    remote_file = remote_files[file.full_path]
+                    file_slot_warnings, file_slot_errors = (
+                        _validate_slot_alignment_and_import(
+                            file, remote_file, annotation_format
```

Comment:
A note here that I initially wanted to perform all slot <-> annotation validation **before** getting to this late stage of the import. However, this isn't possible because:
- Early on in the import, we have access to the `local_files` and `remote_files` variables that allow us to check for warnings and errors
- However, if we find any annotations imported to multi-channeled items that we need to adjust so that they are pointed at the base slot, we run into a problem. If at that stage, we make changes to the `local_files` variable (which represents the annotation data to be imported), the changes aren't persisted through to the stage where we perform the imports. This is because the flow of `import_annotations()` is such that we read in local files twice, [presumably to prevent memory issues(](https://github.com/v7labs/darwin-py/blob/v0.8.10/darwin/importer/importer.py#L356)?)

You can see my initial attempt in [this commit](https://github.com/v7labs/darwin-py/pull/895/commits/30bcbe9c0a88692cb6e3f22d320824741d4fdd99). I've done some digging, and I'm unable to determine why we need to read the annotations in twice. Would it be better to go back to the original approach in the above commit, and simply pass the annotations from the 1st read all the way through?